### PR TITLE
Bookstore: Feature/add reducer and actions

### PR DESCRIPTION
**Pull request by Megagig**
### **In This Milestone, I:**

- [ ] Added  Redux Toolkit (npm install react-redux @reduxjs/toolkit).

- [ ] Created a directory that contains all the Redux logic (/src/redux)

- [ ] Configured a Redux store (/src/redux/store.js)

- [ ] Define a slice of state for books that:

1. Includes an array of books (initial state: empty array)
2. Includes a reducer that adds a book
3. Includes a reducer that removes a book

- [ ] Define a slice of state for categories that:

1. Includes an array of categories (initial state: empty array)
2. Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

- [ ] Structured the application files using a "feature folder" approach and the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for the Redux files. The files/folder structure looks like this:

/src
    |---/components
    |---/redux
        |--- /books
                | booksSlice.js
        |--- /categories
                | categoriesSlice.js
        | store.js
    | index.js

### CODE REVIEW

File: src/redux/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createSlice } from '@reduxjs/toolkit';
+const initialState = [];
+const bookSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => [...state, action.payload],
+    removeBook: (state, action) =>
+      state.filter((book) => book.id !== action.payload),
+  },
```
Comment (thecodechaser) [inline src/redux/bookSlice.js]: - [ ] Please make sure to place both reducers in the respective books, and categories folders in the redux directory to follow the duck patterns.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Megagig👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/bookSlice.js (hunk 1 above)
Comment (Megagig) [reply src/redux/bookSlice.js]: @thecodechaser ,
Thank you for reminding me, this has been sorted out.

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @Megagig 🙋‍♀️

Great work implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Duck pattern is followed, perfect!
-  ✅ Great work using the right Git Flow 
-  ✅ Your README looks good, well done!
-  ✅ Linters are working properly, great job! 

## Optional suggestions ❕

 - [OPTIONAL] Kindly consider fixing the `issues page` link in **_Contributing_** section for providing users a good experience and making your README more professional ⛏🤓
 
 
  image


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**